sew.lic

```diff
@@ -298,9 +304,90 @@ class Sew
         tool = 'sewing needles'
       end
     end
+    magic_routine
     do_sew(next_command, tool, lighten)
   end
 
+  def magic_routine
+    return unless @training_spells
+
+    prepare_spell unless @preparing
+    cast_spell if Flags['magic-ready']
+  end
+
+  def cast_spell
+    return unless Flags['magic-ready']
+    return if mana < 40
+
+    unless @settings.cambrinth_items[0]['name']
+      @settings.cambrinth_items = [{
+        'name' => @settings.cambrinth,
+        'cap' => @settings.cambrinth_cap,
+        'stored' => @settings.stored_cambrinth
+      }]
+    end
+    if check_to_harness(@settings.use_harness_when_arcana_locked) && !force_cambrinth
+      harness_mana(@data['cambrinth'].flatten)
+    else
+      @settings.cambrinth_items.each_with_index do |item, index|
+        case @data['cambrinth'].first
+        when Array
+          find_charge_invoke_stow(item['name'], item['stored'], item['cap'], @settings.dedicated_camb_use, @data['cambrinth'][index], @settings.cambrinth_invoke_exact_amount)
+        when Fixnum, Integer
+          find_charge_invoke_stow(item['name'], item['stored'], item['cap'], @settings.dedicated_camb_use, @data['cambrinth'], @settings.cambrinth_invoke_exact_amount)
+        end
+      end
+    end
+
+    cast?(@data['cast'], @data['symbiosis'], @data['before'], @data['after'])
+    Flags.reset('magic-ready')
+    @preparing = false
+  end
+
+  def prepare_spell
+    return if Flags['magic-ready']
+
+    @preparing = true
+    needs_training = %w[Warding Utility Augmentation]
```

Comment:
I'm not sure if we want to allow Sorcery here. Thoughts?

Reply:
I'm thinking no, unless you want to deal with the logic to get blown-off hands fixed as well

Reply:
Also, there's potential for lost crafting tools if both hands go away

Reply:
The main issue is that if you're crafting in town, which is probably something many Traders are doing, it's illegal to use Sorcery. So, another compounded issue.

Reply:
I'm on board with not allowing Sorcery.